From 086440fde95d840ad6b8aadadb0370b263b9809c Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jdapena@igalia.com>
Date: Wed, 29 Jun 2022 15:48:45 +0200
Subject: GCC: disable ARM control flow integrity

When not using Clang, ARM64 control flow integrity should also be
disabled, as it breaks the build.

Bug: 819294
Change-Id: Ic349b1fa5ee1a46835771270d1bf1fb8782912e4
---
 build/config/arm.gni | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build/config/arm.gni b/build/config/arm.gni
index 20da41efbfbc6..e45329b13f867 100644
--- a/build/config/arm.gni
+++ b/build/config/arm.gni
@@ -132,7 +132,7 @@ if (current_cpu == "arm" || v8_current_cpu == "arm") {
     arm_control_flow_integrity = "standard"
 
     # TODO(cavalcantii): enable the feature for the following OSes next.
-    if (is_mac || is_chromeos || is_fuchsia || is_win ||
+    if (!is_clang || is_mac || is_chromeos || is_fuchsia || is_win ||
         target_cpu != "arm64") {
       # target_cpu != "arm64" covers some cases (e.g. the ChromeOS x64 MSAN
       # build) where the target platform is x64, but V8 is configured to use
-- 
2.34.1

